File: ios/SandboxReactNativeViewManager.m
//
//  SandboxReactNativeViewManager.h
//  MultiInstancePOC
//
//  Created by Aliaksandr Babrykovich on 22/06/2025.
//

#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(SandboxReactNativeViewManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(contextId, NSString)
RCT_EXPORT_VIEW_PROPERTY(jsBundleName, NSString)
RCT_EXPORT_VIEW_PROPERTY(moduleName, NSString)
RCT_EXPORT_VIEW_PROPERTY(initialProperties, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(launchOptions, NSDictionary)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)

@end

File: packages/react-native-multinstance/ios/SandboxReactNativeViewManager.swift
//
//  SandboxReactNativeViewManager.swift
//  MultiInstancePOC
//
//  Created by Aliaksandr Babrykovich on 18/06/2025.
//

import Foundation
import React

@objc(SandboxReactNativeViewManager)
class SandboxReactNativeViewManager: RCTViewManager {
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }

  override func view() -> UIView! {
    return SandboxReactNativeView()
  }

  @objc(postMessage:message:)
  func postMessage(_ reactTag: NSNumber, message: NSDictionary) {
    DispatchQueue.main.async { [weak self] in
      guard let self else {
        return
      }

      let view = self.bridge.uiManager.view(forReactTag: reactTag)

      guard let sandboxView = view as? SandboxReactNativeView else {
        return
      }

      sandboxView.postMessage(message as! [AnyHashable : Any])
    }
  }
}
